build_tools/test_frontend.sh

```diff
@@ -17,10 +17,16 @@ everware-server ${OPTS} > $LOG 2>&1 &
 HUB_PID=$!
 sleep 3
 
+if [ `pgrep -f jupyterhub` == "" ] ; then echo "Error starting" && exit 1 ; fi
+
 echo "Start running frontend tests"
-[ -z "$UPLOADDIR" ] && echo "no UPLOADDIR defined" && exit 1
+if [ -z "$UPLOADDIR" ] ; then 
+	echo "no UPLOADDIR defined" 
+	kill ${HUB_PID}
+	exit 1
+fi
 [ -d $UPLOADDIR ] && rm -rf $UPLOADDIR/*
-nose2 -v --start-dir frontend_tests || FAIL=1
+nose2 -v -N $NPROC --start-dir=frontend_tests || FAIL=1
```

Comment:
From lack of comment I assume you do want to run multpile processes. Is this a good idea? It makes it much harder to tell why a test is failing/first thing you do is disable multi-processing to figure out why things are failing. Maybe what we should do is have some tests for basics that run one-at-a-time and then do some afterwards that run in parallel to test that things work for more than one simultaneous user